### ISSUE TITLE
Rise of Tiamat all of the dragon masks 

### ISSUE THREAD

**Issue by Dragon5e** [Sklore/HL_DD_5e_Colab#1011]
This did not give the correct AC. After looking at how Mariner feat gave bonus to unarmored. I now use the following code
      field[abValue].value += #attrmod[aCHA]
     foreach pick in hero from BaseArmor where "( ArmorClass.Heavy | ArmorClass.Shield ) & ( Helper.CurrArmor | Helper.CurrShield )"
          field[abValue].value = 0
     nexteach
     hero.childfound[ArmorClass].field[Bonus].value += field[abValue].value
   
The bonus is 0 if the armor is present or a shield. If not add it as a bonus.

**Comment by Dragon5e** [comment]
Oops Mariner allow light armor so I added all of the armors to exclude.
      doneif (tagis[Helper.ShowSpec] = 0)
      doneif (tagis[Helper.Disable] <> 0)    
      field[abValue].value += #attrmod[aCHA]
     foreach pick in hero from BaseArmor where "( ArmorClass.Heavy | ArmorClass.Medium | ArmorClass.Light | ArmorClass.Shield ) & ( Helper.CurrArmor | Helper.CurrShield )"
          field[abValue].value = 0
     nexteach
     hero.childfound[ArmorClass].field[Bonus].value += field[abValue].value

**Comment by Dragon5e** [comment]
Tested on a monk druid multiclass in wild shape. I believe it's using the bear's charisma and not using the player hmm

**Comment by Dragon5e** [comment]
Ok I have a solution for the druid wildshape. I compared the charisma of the master and the wildshape.
      doneif (tagis[Helper.ShowSpec] = 0)
      doneif (tagis[Helper.Disable] <> 0)    
      field[abValue].value += maximum(master.child[aCHA].field[aModBonus].value,#attrmod[aCHA])
     foreach pick in hero from BaseArmor where "( ArmorClass.Heavy | ArmorClass.Medium | ArmorClass.Light | ArmorClass.Shield ) & ( Helper.CurrArmor | Helper.CurrShield )"
          field[abValue].value = 0
     nexteach
     hero.childfound[ArmorClass].field[Bonus].value += field[abValue].value

**Comment by Dragon5e** [comment]
The bonus was added even when not equipped so please add this code. 
   
      doneif (field[gIsEquip].value = 0)
      doneif (field[gIsAttuned].value = 0)

**Comment by Dragon5e** [comment]
doneif (tagis[Helper.ShowSpec] = 0)
doneif (tagis[Helper.Disable] <> 0)
doneif (field[gIsEquip].value = 0)
  doneif (field[gIsAttuned].value = 0)
field[abValue].value += maximum(master.child[aCHA].field[aModBonus].value,#attrmod[aCHA])
foreach pick in hero from BaseArmor where "( ArmorClass.Heavy | ArmorClass.Medium | ArmorClass.Light | ArmorClass.Shield ) & ( Helper.CurrArmor | Helper.CurrShield )"
field[abValue].value = 0
nexteach
hero.childfound[ArmorClass].field[Bonus].value += field[abValue].value

Reminder you need this code on the other eval scripts or they get all the bonuses free. 
doneif (field[gIsEquip].value = 0)
  doneif (field[gIsAttuned].value = 0)

**Comment by Dragon5e** [comment]
HoDQ have a black mask but the code did not work on it. After thinking about it, I made a duplicate black mask from ROT and overwrite the HoDQ black mask. It work fine now and also got the bootstraps in as well.

**Comment by Fenris447** [comment]
I've got your improved code in. I left out the Wild Shape stuff; based on the description of Wild Shape, it's very much up to the DM whether any item like this would benefit the shape. So I err on the side of *not* automatically giving it to the shape. Adjustments and such can always be used to add it back in if the player and DM agree to that.

I also deprecated the old HotDQ Black Dragon Mask, treating the one in RoT as the replacement/default.

Thanks for all these fixes!
